aredis_om/model/model.py

```diff
@@ -728,10 +765,27 @@ def resolve_redisearch_query(cls, expression: ExpressionOrNegated) -> str:
         return result
 
     async def execute(self, exhaust_results=True, return_raw_result=False):
-        args = ["ft.search", self.model.Meta.index_name, self.query, *self.pagination]
+        args: List[Union[str, bytes]] = [
+            "FT.SEARCH",
+            self.model.Meta.index_name,
+            self.query,
+            *self.pagination,
+        ]
         if self.sort_fields:
             args += self.resolve_redisearch_sort_fields()
 
+        if self.query_params:
+            args += ["PARAMS", str(len(self.query_params))] + self.query_params
+
+        if self.knn:
+            # Ensure DIALECT is at least 2
+            if "DIALECT" not in args:
+                args += ["DIALECT", "2"]
+            else:
+                i_dialect = args.index("DIALECT") + 1
+                if int(args[i_dialect]) < 2:
```

Comment:
There are 4 dialects. IMHO != 2 if we want to force 2 here.

Reply:
KNN is supported in dialect 2 *and up*: https://redis.io/docs/stack/search/reference/vectors/#querying-vector-fields

Reply:
bump

Reply:
Correct - my statement is about forcing dialect 2 vs 4. Clearly not 1, agreed. The reality is this is output based, and nothing else, it's a nit.

Reply:
My goal was to ensure a compatible dialect is used, it doesn't have to be dialect 2.